### PR TITLE
Added income test to WFF family tax credit

### LINKED ISSUE #91: WFF missing income thresholds
currently the WFF rules don't include the requirement for being below an income threshold... All WFF benefits (In work tax credit, minimum family tax credit and family tax credit) have income thresholds, so we definitely need criteria in all 3 rules that checks that their income is below the income threshold.

### PR DESCRIPTION
**Pull request by Br3nda**
* Tax and benefit system evolution.
* Impacted periods:  1 April 2017 to 31 March 2019
* Impacted areas: `parameters/entitlements/income_tax/family_scheme/family_tax_credit/full_year_abatement_threshold.yaml`

* Details:
Income  test was missing from Family Tax Credit formula

closes #91

### CODE REVIEW

File: openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py [hunk 1]
```diff
@@ -12,7 +12,12 @@ class family_scheme__qualifies_for_family_tax_credit(Variable):
     reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/DLM1518484.html"
 
     def formula(persons, period, parameters):
-        return persons("family_scheme__base_qualifies", period)
+        family_income = persons.family.sum(persons.family.members(
+            "family_scheme__assessable_income", period.this_year))
+        threshold = parameters(
+            period).entitlements.income_tax.family_scheme.family_tax_credit.full_year_abatement_threshold
+        income_under_threshold = family_income < threshold
```
Comment (verbman) [inline openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py]: We should create another variable to do the calculation involving the abatement threshold, I believe the calculation is somewhat more complicated and will look at it now.

File: openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml [hunk 2]
```diff
@@ -26,4 +26,47 @@
...
+    - id: "Whanau"
+      principal_caregiver: "Papatūānuku"
+      partners: ["Rakinui"]
+  titled_properties:
+    - id: "Whare"
+      others: ["Papatūānuku", "Rakinui"]
+  output_variables:
+    family_scheme__qualifies_for_family_tax_credit:
+      2018-08:
+        - false
```
Comment (verbman) [inline openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml]: A test with family_scheme__assessable_income that passes the family_scheme__qualifies_for_family_tax_credit is probably a good idea also.

File: openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py (hunk 1 above)
Comment (verbman) [reply openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py]: This is the formula: http://www.legislation.govt.nz/act/public/2007/0097/latest/DLM1518512.html

File: openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml (hunk 2 above)
Comment (Br3nda) [reply openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml]: There's already a passing test in the file. I added the test for when they're not eligible

Comment (Br3nda) [reply openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml]: I'm not sure i understand your comment. You want a test where we pass in a value for `family_scheme__qualifies_for_family_tax_credit`?

File: openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py (hunk 1 above)
Comment (Br3nda) [reply openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py]: I agree we can add the calculation later, but smart start need only the boolean of qualifying, and not having the income test is blocking their work.

File: openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml (hunk 2 above)
Comment (verbman) [reply openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml]: My bad, I didn't realise the first test was passing in `family_scheme__assessable_income` values, I will mark this resolved.

File: openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py (hunk 1 above)
Comment (alexey0511) [reply openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py]: can we have a variable something like   ...below_threshold?

Comment (Br3nda) [reply openfisca_aotearoa/variables/acts/income_tax/family_scheme/family_tax_credit.py]: We've done more research and there isn't a threshold for WFF. you can earn millions if you have enough dependents.
Instead we are now implementing the actual calculation, and we'll also make a boolean variable that tells you if they qualify for more than $0.00

**Pull request merged**